### PR TITLE
demo(tabset): added more demos

### LINKED ISSUE #378: Tabs: add more demos
The ones I can think of are:
* selecting by id
* preventing tab selection on `change` event

We will probably identify more based on users feedback / requests.

### PR DESCRIPTION
**Pull request by sendilkumarn**
Before submitting a pull request, please make sure you have at least performed the following:

 - [X] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/core/blob/master/CONTRIBUTING.md) guide.
 - [X] built and tested the changes locally.
 - [X] added/updated any applicable tests.
 - [X] added/updated any applicable API documentation.
 - [X] added/updated any applicable demos.

fixes #378

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
It needs rebasing (doesn't merge cleanly). Also you shouldn't be changing existing convention on how things are imported (!)

**Comment by sendilkumarn** [comment]
just noted. that import was changed when I was working I guess. Changed it.

File: demo/src/app/components/tabset/demos/basic/index.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export * from  './tabset-basic';
+
+export const basicTsContent = require('!!prismjs?lang=typescript!./tabset-basic');
+export const basicHtmlContent = require('!!prismjs?lang=markup!./tabset-basic.html');
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/demos/basic/index.ts]: This is not a good change. We shouldn't have `index.ts` file in this folder.

Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/demos/basic/index.ts]: Delete this file.

File: demo/src/app/components/tabset/demos/basic/tabset-basic.ts [hunk 2]
```diff
@@ -6,5 +6,4 @@ import {NGB_TABSET_DIRECTIVES} from '@ng-bootstrap/ng-bootstrap';
   template: require('./tabset-basic.html'),
   directives: [NGB_TABSET_DIRECTIVES]
 })
-export class NgbdTabsetBasic {
-}
+export class TabsetBasicComponent { }
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/demos/basic/tabset-basic.ts]: Why this change? It is not good. Please revert.

File: demo/src/app/components/tabset/demos/index.ts [hunk 3]
```diff
@@ -1,9 +1,4 @@
-import {NgbdTabsetBasic} from './basic/tabset-basic';
-
-export const DEMO_DIRECTIVES = [NgbdTabsetBasic];
-
-export const DEMO_SNIPPETS = {
-  "basic": {
-    "code": require('!!prismjs?lang=typescript!./basic/tabset-basic'),
-    "markup": require('!!prismjs?lang=markup!./basic/tabset-basic.html')}
-};
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/demos/index.ts]: This change is not good. Please add your imports from code / markup here instead of putting them in index.ts of individual demos.

File: demo/src/app/components/tabset/demos/pills/index.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export * from './tabset-pills';
+
+export const pillsTsContent = require('!!prismjs?lang=typescript!./tabset-pills');
+export const pillsHtmlContent = require('!!prismjs?lang=markup!./tabset-pills.html');
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/demos/pills/index.ts]: As above. We shouldn't have this index.ts file

File: demo/src/app/components/tabset/demos/preventChange/index.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export * from './tabset-preventChange';
+
+export const preventChangeTsContent = require('!!prismjs?lang=typescript!./tabset-preventChange');
+export const preventChangeHtmlContent = require('!!prismjs?lang=markup!./tabset-preventChange.html');
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/demos/preventChange/index.ts]: As above - this index.ts file shouldn't exist

File: demo/src/app/components/tabset/demos/selectById/index.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export * from './tabset-selectById';
+
+export const selectByIdTsContent = require('!!prismjs?lang=typescript!./tabset-selectById');
+export const selectByIdHtmlContent = require('!!prismjs?lang=markup!./tabset-selectById.html');
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/demos/selectById/index.ts]: As above - this index.ts file shouldn't exist

File: demo/src/app/components/tabset/tabset.component.ts [hunk 7]
```diff
@@ -1,23 +1,31 @@
...
+    ContentWrapper,
+    NgbdApiDocs,
+    TabsetBasicComponent,
+    TabsetPillsComponent,
+    TabsetSelectByIdComponent,
+    TabsetPreventChangeComponent,
+    ExampleBoxComponent]
 })
 export class NgbdTabs {
-   snippets = DEMO_SNIPPETS;
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/tabset.component.ts]: You should not delete this line.

File: demo/src/app/components/tabset/tabset.component.ts [hunk 8]
```diff
@@ -1,23 +1,31 @@
...
+  basicHtmlContent = basicHtmlContent;
+  basicTsContent = basicTsContent;
+  pillsHtmlContent = pillsHtmlContent;
+  pillsTsContent = pillsTsContent;
+  selectByIdHtmlContent = selectByIdHtmlContent;
+  selectByIdTsContent = selectByIdTsContent;
+  preventChangeHtmlContent = preventChangeHtmlContent;
+  preventChangeTsContent = preventChangeTsContent;
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/tabset/tabset.component.ts]: You shouldn't be adding those lines - everything should be part of `DEMO_SNIPPETS` already

**Comment by icfantv** [comment]
I'm not entirely convinced we need 4 demos for tabs right now.  The API is VERY simple at the moment and the only other thing I can see demoing is the `preventDefault()` which can be just another tab in the existing demo.

Once the forms API is complete, I would like to do another forms demo like we do for UIBS as this is a very common use case for users.

**Comment by pkozlowski-opensource** [comment]
So, the demo itself looks OK but the files structure is not. Please see my comments. You can see how demos are structured in, say, https://github.com/ng-bootstrap/ng-bootstrap/tree/master/demo/src/app/components/tooltip/demos

**Comment by sendilkumarn** [comment]
@pkozlowski-opensource alerts / accordion / button were taken as reference, all were like this.

I initially did that way and then changed entirely so that it align with alerts / accordion and button.

**Comment by pkozlowski-opensource** [comment]
> alerts / accordion / button / collapse were taken as reference, all were like this.
> I initially did that way and then changed entirely so that it align with alerts / accordion / collapse and button

Yeh, sorry about this, we've got a mix of styles right now :-/ But you should follow one that we've got in tooltips / popovers. We even got a script that generates skeletons of those demos, see: https://github.com/ng-bootstrap/ng-bootstrap/blob/master/misc/demo-gen.js

Once again, I'm really sorry that it wasn't clear. I will clean up all the demos after the release so we are consistent. 

Next time don't hesitate to ask if in doubt.

**Comment by sendilkumarn** [comment]
I will change others if it is okay

**Comment by pkozlowski-opensource** [comment]
Reworked it slightly and merged. Thnx!
